kernels/ZendEngine3/extended/fcall.c

```diff
@@ -1197,9 +1197,12 @@ int zephir_call_function_opt(zend_fcall_info *fci, zend_fcall_info_cache *fci_ca
 
 #if PHP_VERSION_ID >= 70100
 	//EG(fake_scope) = calling_scope;
-        if(fci_cache->object) {
-            Z_OBJ(call->This) = fci_cache->object;
-        }
+	if (fci_cache->object) {
+		Z_OBJ(call->This) = fci_cache->object;
+	} else {
+		Z_OBJ(call->This) = fci->object;
```

Comment:
Fllow `zend_get_called_scope`,  If there is no `object` then use `ce` set to `This`
```c
ZEND_API zend_class_entry *zend_get_called_scope(zend_execute_data *ex) /* {{{ */
{
	while (ex) {
		if (Z_TYPE(ex->This) == IS_OBJECT) {
			return Z_OBJCE(ex->This);
		} else if (Z_CE(ex->This)) {
			return Z_CE(ex->This);
		} else if (ex->func) {
			if (ex->func->type != ZEND_INTERNAL_FUNCTION || ex->func->common.scope) {
				return NULL;
			}
		}
		ex = ex->prev_execute_data;
	}
	return NULL;
}
``